modules/service/src/main/scala/lucuma/odb/graphql/mapping/QueryMapping.scala

```diff
@@ -62,45 +60,44 @@ trait QueryMapping[F[_]] extends Predicates[F] {
     path:     Path,
     pid:      model.Program.Id,
     oid:      model.Observation.Id,
-    useCache: Boolean
   ): F[Result[Json]] =
-    (for {
-      p <- EitherT(services.useTransactionally(itc(itcClient).selectParams(pid, oid)))
-      r <- EitherT(services.useNonTransactionally(itc(itcClient).callService(p, useCache)))
-    } yield r).value.map {
-      case Left(errors)     => Result.failure(errors.map(_.format).intercalate(", "))
-      case Right(resultSet) => resultSet.asJson.success
+    services.useNonTransactionally {
+      itcService(itcClient)
+        .lookup(pid, oid)
+        .map {
+          case Left(e)  => Result.failure(e.format)
+          case Right(s) => s.result.asJson.success
+        }
     }
 
   // TODO: this is used by a now deprecated part of the schema.  Remove when possible.
   def sequence(
     path:        Path,
     pid:         model.Program.Id,
     oid:         model.Observation.Id,
-    useCache:    Boolean,
     futureLimit: Generator.FutureLimit
   ): F[Result[Json]] = {
 
-    val mapResult: Generator.Result => Result[Json] = {
-      case Generator.Result.ObservationNotFound(_, _) => Result(Json.Null)
-      case e: Generator.Error                         => Result.failure(e.format)
-      case Generator.Result.Success(_, itc, exec, d)  =>
-          Result(Json.obj(
-            "programId"       -> pid.asJson,
-            "observationId"   -> oid.asJson,
-            "itcResult"       -> itc.asJson,
-            "executionConfig" -> exec.asJson,
-            "scienceDigest"   -> d.asJson
-          ))
+    val mapError: Generator.Error => Result[Json] = {
+      case Generator.Error.ItcError(ItcService.Error.ObservationNotFound(_, _)) =>
+        Result(Json.Null)
+      case e: Generator.Error                                                   =>
+       Result.failure(e.format)
     }
 
-    services.use { s =>
-      val generatorService = s.generator(commitHash, itcClient, plannedTimeCalculator)
+    def mapSuccess(s: Generator.Success): Result[Json] =
+      Json.obj(
+        "programId"       -> pid.asJson,
+        "observationId"   -> oid.asJson,
+        "itcResult"       -> s.itc.asJson,
+        "executionConfig" -> s.value.asJson,
+        "scienceDigest"   -> s.scienceDigest.asJson
+      ).success
 
-      (for {
-        p <- EitherT(s.transactionally(generatorService.selectParams(pid, oid)))
-        r <- EitherT(generatorService.generate(oid, p, useCache, futureLimit))
-      } yield r).value.map(result => mapResult(result.merge))
+    services.use { s =>
+      s.generator(commitHash, itcClient, plannedTimeCalculator)
+       .generate(pid, oid, futureLimit)
+       .map(_.bimap(mapError, mapSuccess).merge)
```

Comment:
All of these updates in `QueryMapping` are duplicated code but they should be removed when we remove the corresponding deprecated schema bits.